### PR TITLE
[#41480] respond with 401 on invalid bearer tokens

### PR DESCRIPTION
**Pull request by Kharonus**
- https://community.openproject.org/work_packages/41480
- amend strategy for doorkeeper to react on all bearer tokens
- added ouath request spec for valid, invalid and revoked token

### CODE REVIEW

File: lib/open_project/authentication/strategies/warden/doorkeeper_oauth.rb [hunk 1]
```diff
@@ -9,15 +9,20 @@ module Warden
         #
         class DoorkeeperOAuth < ::Warden::Strategies::Base
           def valid?
-            @token = ::Doorkeeper::OAuth::Token.authenticate(decorated_request, *Doorkeeper.configuration.access_token_methods)
-            @token&.accessible? && @token&.acceptable?(scope)
+            bearer_token = ::Doorkeeper::OAuth::Token.from_request(decorated_request,
+                                                                   *Doorkeeper.configuration.access_token_methods)
+            bearer_token != nil
```
Comment (ulferts) [inline lib/open_project/authentication/strategies/warden/doorkeeper_oauth.rb]: can be rewritten as:

```
::Doorkeeper::OAuth...
  .from_request(...)
  .present?
```

**Review by ulferts — APPROVED** [review]
I did not vet the changes to the docker setup but the OAuth changes look solid.

File: lib/open_project/authentication/strategies/warden/doorkeeper_oauth.rb (hunk 1 above)
Comment (Kharonus) [reply lib/open_project/authentication/strategies/warden/doorkeeper_oauth.rb]: indeed O.O changed, of course

File: docker-compose.yml [hunk 2]
```diff
@@ -94,6 +96,8 @@ services:
     stop_grace_period: "3s"
     volumes:
       - "pgdata:/var/lib/postgresql/data"
+    ports:
+      - "54322:5432"
```
Comment (b12f) [inline docker-compose.yml:100]: You can probably even open this up on the default port

**Review by b12f — APPROVED** [review]
LGTM

File: docker-compose.yml (hunk 2 above)
Comment (Kharonus) [reply docker-compose.yml:100]: I cannot, as the `db-test` opens the default port. Also it kind of clashes of course with any native postgres dbs.

I could change the `db-test` to an alternative port, change this to default and be specific about, that you cannot run another postgres db on default port, when launching standard compose. You have then to alter either your compose service, or your running db to make it work.

What do you think?

**Comment by ulferts** [comment]
Merging as the failing specs are already on `dev`

**Pull request merged**